scipy/optimize/cython_optimize/__init__.py

```diff
@@ -0,0 +1,630 @@
+                      double xtol, double rtol, int iter, scipy_zeros_parameters *full_output)
+
+     :param callback_type_tuple f: the callback function
+     :param double xa: lower bound
+     :param double xb: upper bound
+     :param tuple args: extra parameters
+     :param double xtol: absolute tolerance
+     :param double rtol: relative tolerance
+     :param int iter: maximum number of iterations
+     :param scipy_zeros_parameters* full_output: optional output ``struct``, use ``NULL`` if not desired
+     :return: double
+
+- ``cython_optimize.zeros_tuple.brenth``
+
+  .. py:function:: scipy.optimize.cython_optimize.zeros_tuple.brenth(f, xa, xb, args, xtol, rtol, iter, full_output)
+
+     Similiar to :py:func:`~scipy.optimize.brenth` with a callback, ``f``, that
+     takes a double and a Python tuple ::
+
+        double brenth(callback_type_tuple f, double xa, double xb, tuple args,
+                      double xtol, double rtol, int iter, scipy_zeros_parameters *full_output)
+
+     :param callback_type_tuple f: the callback function
+     :param double xa: lower bound
+     :param double xb: upper bound
+     :param tuple args: extra parameters
+     :param double xtol: absolute tolerance
+     :param double rtol: relative tolerance
+     :param int iter: maximum number of iterations
+     :param scipy_zeros_parameters* full_output: optional output ``struct``, use ``NULL`` if not desired
+     :return: double
+
+- ``cython_optimize.zeros_tuple.brentq``
+
+  .. py:function:: scipy.optimize.cython_optimize.zeros_tuple.brentq(f, xa, xb, args, xtol, rtol, iter, full_output)
+
+     Similiar to :py:func:`~scipy.optimize.brentq` with a callback, ``f``, that
+     takes a double and a Python tuple ::
+
+        double brentq(callback_type_tuple f, double xa, double xb, tuple args,
+                      double xtol, double rtol, int iter, scipy_zeros_parameters *full_output)
+
+     :param callback_type_tuple f: the callback function
+     :param double xa: lower bound
+     :param double xb: upper bound
+     :param tuple args: extra parameters
+     :param double xtol: absolute tolerance
+     :param double rtol: relative tolerance
+     :param int iter: maximum number of iterations
+     :param scipy_zeros_parameters* full_output: optional output ``struct``, use ``NULL`` if not desired
+     :return: double
+
+Examples
+--------
+Usage of :py:mod:`~scipy.optimize.cython_optimize` requires the use of Cython
+to write callbacks that are compiled into native C. See the Cython Optimize
+Zeros API Jupyter notebook in the ``cython_optimize/Examples`` folder for usage
+examples. There are also several examples that solve the single-diode model of
+a solar cell in the ``cython_optimize/Examples`` folder. These examples are
+also used to test the different combinations of callbacks and root-finders.
+
+These are the basic steps to use :py:mod:`~scipy.optimize.cython_optimize`:
+
+1. Select a callback signature, for example:
```

Comment:
still unclear why 3 signatures, seems needlessly complex?

Reply:
which signature is preferable? `scipy.integrate.quad` has four. If I had to guess which to keep, I guess, I just keep `(double, void*)`, is that the preference?